File: Java/Tests/src/PI/Pi.java
package PI;

public class Pi {

    public static void main(String args[]){

       double y = Math.sqrt(2)-1;
       double a = 6 - 4*Math.sqrt(2);
       double Y,A;
       double pi;
       for(int i = 0 ; i < 100 ; i++){
           Y = (1-Math.pow(1-Math.pow(y,4),0.25))/(1+Math.pow(1-Math.pow(y,4),0.25));
           A = (Math.pow(1+Y,4)*a)-(Math.pow(2,2*i -1)*y*(1+y+Math.pow(y,2)));
           pi = 1/a;
           System.out.println(pi);
       }

    }

}

File: Scala/src/Math/PI.scala
package Math

import java.math.MathContext

import scala.math.ScalaNumber

object PI {

  def sqrt(x : BigDecimal) : BigDecimal = {
    val epsilon = BigDecimal(1E-15)
    var t = x
    var cont1 = BigDecimal(1)
    var cont2 = BigDecimal(1)
    do {
      t = (x / t + t) / BigDecimal(2)
      cont1 = t- x/t
      cont2 = cont1.abs
    }while(cont2 > epsilon)
    t
  }
  def sqrtd(x : BigDecimal) : BigDecimal = {
    val epsilon = BigDecimal(1E-3)
    var t = x
    var cont1 = BigDecimal(1)
    var cont2 = BigDecimal(1)
    do {
      t = ((x / (t*t)) + BigDecimal(2)*t) / BigDecimal(3)
      cont1 = t - x/t
      cont2 = cont1.abs
    }while(cont2 > epsilon)
    t
  }

  def main(args: Array[String]): Unit = {
    /**for(j <- 1 to 10){
      var a = BigDecimal(1)
      var b = a / sqrt(BigDecimal(2))
      var s = BigDecimal(0.25)
      var A, B, c, S = BigDecimal(1)

      for(i <- 0 to j){
        A = (a + b)/BigDecimal(2)
        B = sqrt(a * b)
        c = (a-A)*(a-A)
        S = s - BigDecimal(2).pow(i)*c

        a = A
        b = B
        s = S
      }

      var PI = (A*A)/s
      println(j + ". :" + (PI.toDouble - scala.math.Pi))
    }

      **/

    var a = sqrtd(BigDecimal(27))
    print(a)

  }
}
